### PR TITLE
Delete content from publishing API on discard.

### PR DESCRIPTION
**Pull request by elliotcm**
Reissuing https://github.com/alphagov/whitehall/pull/2400 against master.

https://trello.com/c/sPUPz6FN/436-have-whitehall-delete-discarded-drafts

### CODE REVIEW

File: lib/whitehall/publishing_api.rb [hunk 1]
```diff
@@ -55,6 +55,10 @@ def self.publish_gone(base_path)
       PublishingApiGoneWorker.perform_async(base_path)
     end
 
+    def self.discard_draft_async(edition)
+      PublishingApiDiscardDraftWorker.perform_async(edition.content_id, edition.translations.map(&:locale))
```
Comment (danielroseman) [inline lib/whitehall/publishing_api.rb]: This uses a different pattern to the other methods in this class, which map over `locales_for` and enqueue a job for each locale. Unless there is a good reason to do it differently, we should be consistent (and if there is a good reason, we should document it).

Comment (elliotcm) [inline lib/whitehall/publishing_api.rb]: We wanted the worker to work on all the locales at once, so as to try to keep them at least a little bit atomic.  It's a debatable opinion and I'm not sure it really warrants either consistency or documentation as it depends on the sort of update you're doing (deletes are fast and not easily prone to error).

Comment (elliotcm) [inline lib/whitehall/publishing_api.rb]: It's also an idempotent operation so it doesn't matter if it's repeated.

Comment (danielroseman) [inline lib/whitehall/publishing_api.rb]: Fair enough.

**Comment by danielroseman** [comment]
:+1:

**Pull request merged**